sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DeduplicateRelations.scala

```diff
@@ -38,28 +38,31 @@ case class RelationWrapper(cls: Class[_], outputAttrIds: Seq[Long])
 object DeduplicateRelations extends Rule[LogicalPlan] {
   override def apply(plan: LogicalPlan): LogicalPlan = {
     val newPlan = renewDuplicatedRelations(mutable.HashSet.empty, plan)._1
-    if (newPlan.find(p => p.resolved && p.missingInput.nonEmpty).isDefined) {
-      // Wait for `ResolveMissingReferences` to resolve missing attributes first
-      return newPlan
+
+    def containsMissingInput(p: LogicalPlan) = {
```

Comment:
nit: How about?

```scala
def noMissingInput(p: LogicalPlan) = !p.exists(_.missingInput.nonEmpty)
```

Reply:
Thanks, that's a good idea. I crafted the commit on an older versions where `.exists()` was not available, but will update the PR soon.

Reply:
Fixed in https://github.com/apache/spark/pull/46183/commits/e47bfb94c7494b10c9cd8b48e4b377d0a39dd445.